test/unix/main.ml

```diff
@@ -107,16 +149,68 @@ let test_add t =
       if not (List.mem v l && List.mem v' l) then
         Alcotest.fail (Printf.sprintf "Adding duplicate values failed.")
 
+let test_find_present_lwt t =
+  Lwt_list.iter_p
+    (fun (k, v) ->
+      Lwt.return
+        ( match Index.find_all t k with
+        | [] ->
+            Alcotest.fail
+              (Printf.sprintf "Wrong insertion: %s key is missing." k)
+        | l ->
+            if not (List.mem v l) then
+              Alcotest.fail
+                (Printf.sprintf "Wrong insertion: %s value is missing." v) ))
+    !l
+
 let find_present_live () = test_find_present t
 
 let find_absent_live () = test_find_absent t
 
+let no_extra_values_added () = test_no_extra_values_added t
+
+let membership () = test_mem t
+
+let find_present_live_lwt () = Lwt_main.run (test_find_present_lwt t)
```

Comment:
```suggestion
let find_present_live_lwt _ () = test_find_present_lwt t
```

Use `Alcotest_lwt.test_case` for tests inside the Lwt monad to avoid needing to call `Lwt_main.run`. You'd also need to avoid the Lwt_switch.t that it offers you.